docker/daemon.go

```diff
@@ -264,7 +264,9 @@ func (cli *DaemonCli) CmdDaemon(args ...string) error {
 	cli.TrustKeyPath = commonFlags.TrustKey
 
 	registryService := registry.NewService(cli.Config.ServiceOptions)
-	d, err := daemon.NewDaemon(cli.Config, registryService)
+
+	containerdRemote := cli.getContainerdRemote()
```

Comment:
this isn't returning an error (just log it) and when dereferencing in NewDaemon it panics (I guess)